py/picca/delta_extraction/expected_fluxes/true_continuum.py

```diff
@@ -0,0 +1,486 @@
+                                      fill_value="extrapolate")
+        self.get_mean_cont_weight = interp1d(log_lambda_cont,
+                                             mean_cont_weight,
+                                             fill_value=0.0,
+                                             bounds_error=False)
+
+
+    def populate_los_ids(self, forests):
+        """Populate the dictionary los_ids with the mean expected flux, weights,
+        and inverse variance arrays for each line-of-sight.
+
+        Arguments
+        ---------
+        forests: List of Forest
+        A list of Forest from which to compute the deltas.
+        """
+
+        for forest in forests:
+            if forest.bad_continuum_reason is not None:
+                self.logger.info(f"Rejected forest with los_id {forest.los_id} "
+                                 f"due to {forest.bad_continuum_reason}")
+                continue
+
+            # get the variance functions and statistics
+            if Forest.wave_solution == "log":
+                var_lss = self.get_var_lss(forest.log_lambda)
+            elif Forest.wave_solution == "lin":
+                var_lss = self.get_var_lss(forest.lambda_)
+            else:
+                raise ExpectedFluxError("Forest.wave_solution must be either "
+                                        "'log' or 'lin'")
+
+            mean_expected_flux = forest.continuum
+            var_pipe = 1. / forest.ivar/ forest.continuum**2
+            variance =  var_lss + var_pipe
+            weights = 1. / variance
+
+            if isinstance(forest, Pk1dForest):
+                ivar = forest.ivar / mean_expected_flux**2
+
+                self.los_ids[forest.los_id] = {
+                    "mean expected flux": mean_expected_flux,
+                    "weights": weights,
+                    "ivar": ivar,
+                    "continuum": forest.continuum,
+                }
+            else:
+                self.los_ids[forest.los_id] = {
+                    "mean expected flux": mean_expected_flux,
+                    "weights": weights,
+                    "continuum": forest.continuum,
+                }
+
+    def save_delta_attributes(self,out_dir):
+        """Save mean continuum in the delta attributes file
+
+        Arguments
+        ---------
+
+        out_dir: str
+        Directory where data will be saved
+
+        Raise
+        -----
+        ExpectedFluxError if Forest.wave_solution is not 'lin' or 'log'
+        """
+        iter_out_file = self.iter_out_prefix + ".fits.gz"
+
+        with fitsio.FITS(out_dir + iter_out_file, 'rw',
+                         clobber=True) as results:
+            header = {}
+            header["FITORDER"] = -1
+            if Forest.wave_solution == "log":
+                num_bins = int((Forest.log_lambda_max - Forest.log_lambda_min) /
+                               Forest.delta_log_lambda) + 1
+                stack_log_lambda = (
```

Comment:
This variable "stack_log_lambda" is not used anywhere, is it?

Reply:
Ah, is a leftover from when I was saving it, thanks for noticing. Will remove it.